TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt

```diff
@@ -41,7 +42,32 @@ data class Click(
      * The marketplace's ID for the click
      */
     val id: String,
-)
+) {
+    fun toJsonObject(): JSONObject {
+        return JSONObject()
+            .put("resolvedBidId", resolvedBidId)
+            .put("entity", entity?.toJsonObject())
+            .put("additionalAttribution", additionalAttribution)
+            .put("placement", placement.toJsonObject())
+            .put("occurredAt", occurredAt)
+            .put("opaqueUserId", opaqueUserId)
+            .put("id", id)
+    }
+
+    companion object {
+        fun fromJsonObject(json: JSONObject): Click {
+            return Click(
+                resolvedBidId = json.getString("resolvedBidId"),
+                entity = Entity.fromJsonObject(json.getJSONObject("entity")),
+                additionalAttribution = json.getString("additionalAttribution"),
+                placement = Placement.fromJsonObject(json.getJSONObject("placement")),
+                occurredAt = json.getString("occurredAt"),
+                opaqueUserId = json.getString("opaqueUserId"),
+                id = json.getString("id"),
+            )
+        }
+    }
+}
```

Comment:
On which situations would you want to parse a json event? I think that, other than for testing purposes, this is not needed.

Reply:
Current cache implementation writes the serialized json to SharedPreferences (and re-reads it before send the event), I'll move this into test-specific classes when convenient.